### PR TITLE
Strategy design pattern Implementation

### PR DESCRIPTION
**Pull request by JoshJ24**
This pull request features the refactoring of code to implement the strategy design pattern. Essentially in the data access object for dashboard, the methods like update account, add account, retrieve accounts and delete account require encryption and decryption on the data. To do this, encrypt and decrypt methods along with further helper methods converting byte array to a hexadecimal string and vice versa were implemented in the data access object. To improve design with the strategy design pattern, an encryption strategy interface was implemented so that depending on what encryption strategy was set, the encryptions and decryptions would vary. 
Thus the interface encryption strategy was created with methods to encrypt and decrypt, the AES256EncryptionStrategy was added which implements this interface and is the default encryption strategy. Lastly an Encryption Utilities class was added which handles the hexadecimal to byte Array and byte Array to hexadecimal String methods in order to follow the single responsibility principle as well as Interface segregation principle so that the other encryption strategies which may not require these utility methods don't have to implement them. 
In summary, based on the set encryption strategy which is an interface with encrypt and decrypt methods, in the DAO, all the original encrypt and decrypt calls were replaced with EncryptionStrategy.encrypt and decrypt calls where encryption strategy is the set encryption strategy. For this program, so far only AES256 encryption strategy was utilized thus it was set as the default strategy, but other encryption strategies could be added in the future as well.

### CODE REVIEW

**Review by hadi-naqvi — APPROVED** [review]
The implementation of the strategy design pattern was done correctly  and makes for a better code structure as it adheres to the SOLID principles more closely. In particular, the logic for the encryption/decryption was separated from the DAOs which helps for the single responsibility principle. Likewise, the new implementation is open for extension but closed for modification, which is also a part of the SOLID principles. Overall, all the code was documented clearly with javadocs, variable naming conventiosn were good, and the newly added features are correct and the program still functions as it did prior after running some test cases. Great work.
